### PR TITLE
Nomicon: Change `state rent` to `storage staking`

### PR DESCRIPTION
**Pull request by idea404**
This PR changes the outdated concept in `FinancialTransaction.md`. 

Mentioned: https://github.com/near/docs/issues/345

### CODE REVIEW

File: specs/RuntimeSpec/Scenarios/FinancialTransaction.md [hunk 1]
```diff
@@ -110,10 +110,11 @@ If any of the above operations fail all of the changes will be reverted.
 The receipt created in the previous section will eventually arrive to a runtime on the shard that hosts `bob_near` account.
 Again, it will be processed by `Runtime::apply` which will immediately call `Runtime::process_receipt`.
 It will check that this receipt does not have data dependencies (which is only the case of function calls) and will then call `Runtime::apply_action_receipt` on `TransferAction`.
+
 `Runtime::apply_action_receipt` will perform the following checks:
 
 - Retrieves the state of `bob_near` account, if it still exists (it is possible that Bob has deleted his account concurrently with the transfer transaction);
-- Applies the rent to Bob's account;
+- Checks if `bob_near` has enough balance for storage staking;
```
Comment (jakmeier) [inline specs/RuntimeSpec/Scenarios/FinancialTransaction.md:117]: this sounds a bit confusing to me
1. there is no change in storage, so the check is not needed for the example (of course it is still done in reality, just not very relevant to explain how a NEAR token transfer works)
2. In the code, the check is done after everything has been applied to the temporary state. The temporary state is dropped / rolled back if there is not enough balance. Maybe this level of detail is not relevant but still I would consider switching the order in this list, moving the storage staking check at the end.

File: specs/RuntimeSpec/Scenarios/FinancialTransaction.md [hunk 2]
```diff
@@ -110,10 +110,11 @@ If any of the above operations fail all of the changes will be reverted.
...
 - Computes the cost of processing a receipt and a transfer action;
-- Checks if `bob_near` still exists and if it is deposits the transferred tokens;
-- Computes how much reward should be paid to the validators from the burnt gas.
+- Computes how much reward should be burnt as a means to indirectly reward validators
```
Comment (jakmeier) [inline specs/RuntimeSpec/Scenarios/FinancialTransaction.md:119]: Not sure if "indirectly reward validators" is helpful here. It hints at a much larger topic and the terminology of "indirectly rewarding validators" is not well established AFAIK. I think just saying that we burn it is better here.

Also, you removed "deposits the transferred tokens" from the list, why is that? Seems like the most important step of a NEAR transfer action. :)

File: specs/RuntimeSpec/Scenarios/FinancialTransaction.md [hunk 3]
```diff
@@ -110,10 +110,11 @@ If any of the above operations fail all of the changes will be reverted.
...
+- Checks if `bob_near` still exists and if so, deposits the gas reward to the `bob_near` account;
```
Comment (jakmeier) [inline specs/RuntimeSpec/Scenarios/FinancialTransaction.md:120]: yes :+1:

**Review by jakmeier — COMMENTED** [review]
Nice, important to clean up our naming here!

Can you do the same changes to the other list, that in the section **Transaction to receipt**?

> - Applies the storage rent to the alice_near account;
> ...
> - Checks whether after all these operations account has enough balance to passively pay for the rent for the next several blocks (an economical constant defined by Near Protocol). Otherwise account will be open for an immediate deletion, which we do not want;
> ...
> - Computes how much reward should be paid to the validators from the burnt gas.
>

Also, consider rephrasing this section a bit:
> Pre-requisites
>
> ...
> We also need to assume that both Alice and Bob have some number of tokens on their accounts. Alice needs >100 tokens on the account so that she could transfer 100 tokens to Bob, but also Alice and Bob need to have some tokens to pay for the rent of their account -- which is essentially the cost of the storage occupied by the account in the Near Protocol network.

I think *rent* can still be a good analogy. But maybe we should mention here that we call it "storage staking" because we are not actually taking away any tokens, we only keep the staking rewards they missed out on.

**Comment by idea404** [comment]
> I think rent can still be a good analogy. But maybe we should mention here that we call it "storage staking" because we are not actually taking away any tokens, we only keep the staking rewards they missed out on.

I've updated the section mentioning the cost is refundable. I understood yday from Akhi the storage itself once reserved is not being staked. I find that saying the amount is being staked would thus be misleading. `Refundable deposit` would also work by this matter

File: specs/RuntimeSpec/Scenarios/FinancialTransaction.md (hunk 1 above)
Comment (idea404) [reply specs/RuntimeSpec/Scenarios/FinancialTransaction.md]: have updated section, I like to include detail myself, especially here considering it's Nomicon, but if you still feel this does not belong in this run-through I'm happy to remove!

File: specs/RuntimeSpec/Scenarios/FinancialTransaction.md (hunk 2 above)
Comment (idea404) [reply specs/RuntimeSpec/Scenarios/FinancialTransaction.md]: Missed that! Ty!

File: specs/RuntimeSpec/Scenarios/FinancialTransaction.md (hunk 1 above)
Comment (jakmeier) [reply specs/RuntimeSpec/Scenarios/FinancialTransaction.md]: yeah with this item at the end, it's probably good to keep it in

**Comment by jakmeier** [comment]
> > I think rent can still be a good analogy. But maybe we should mention here that we call it "storage staking" because we are not actually taking away any tokens, we only keep the staking rewards they missed out on.
> 
> I've updated the section mentioning the cost is refundable. I understood yday from Akhi the storage itself once reserved is not being staked. I find that saying the amount is being staked would thus be misleading. `Refundable deposit` would also work by this matter

True, we are not staking it in the sense that it is being used for consensus. Somehow we still ended up using "storage staking" as the term for users paying for storage by locking up their tokens. But I agree with your reasoning.

In that sense, maybe we should not call it storage staking at all in this document? Maybe replacing the two occurrences with something like  "reserved balance for storage" would be clearer? But no strong opinion from my side, it's fine to keep it as it is or to change it.

**Review by frol — APPROVED** [review]
I rely on @jakmeier's review here, as he is code-owner for nearcore, so as a moderator, I am approving this change.

@idea404 Please, merge it once you are ready.

**Pull request merged**